log4j-core/src/main/java/org/apache/logging/log4j/core/config/builder/impl/DefaultComponentBuilder.java

```diff
@@ -57,43 +59,51 @@ public DefaultComponentBuilder(final CB builder, final String name, final String
         this.value = value;
     }
 
+    /** {@inheritDoc} */
     @Override
     public T addAttribute(final String key, final boolean value) {
         return put(key, Boolean.toString(value));
     }
 
+    /** {@inheritDoc} */
     @Override
     public T addAttribute(final String key, final Enum<?> value) {
-        return put(key, value.name());
+        return put(key, Optional.ofNullable(value).map(Enum::name).orElse(null));
```

Comment:
```suggestion
        return put(key, value != null ? value.name() : null);
```
We tend to use more primitive expressions in Log4j to keep it garbage-free. For a configuration it doesn't matter, but you'll find a lot of simple `null` checks and iterations on arrays in the code.

Personally I wouldn't use `Optional` for a single method call.